File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    
    let res = Number.MAX_SAFE_INTEGER;
    let j = 0;
    let sum = 0;
    for(let i = 0; i<nums.length; i++){

        sum += nums[i];

        while(sum > target){
            res = Math.min(res,i - j + 1);
            sum -= nums[j++];
        }

         if(sum == target)
            res = Math.min(res,i - j + 1);
    }

    return res == Number.MAX_SAFE_INTEGER ? 0 : res;
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = INT_MAX,j = 0,sum = 0;
        for(int i = 0; i<nums.size(); i++){
            sum += nums[i];
            while(sum > target){
                res = min(res,i - j + 1);
                sum -= nums[j++];
            }

            if(sum == target)
                res = min(res,i - j + 1);
        }
        return res == INT_MAX ? 0 : res;
    }
};